packages/bsync/src/subscriptions/revenueCatWebhookHandler.ts

```diff
@@ -0,0 +1,174 @@
+import { IncomingMessage, ServerResponse } from 'node:http'
+import AppContext from '../context'
+import { isValidDid } from '../routes/util'
+import { Database } from '..'
+import { sql } from 'kysely'
+import { createSubsOpChannel } from '../db/schema/subs_op'
+
+export const isRevenueCatWebhookUrl = (urlStr: string | undefined) => {
+  if (!urlStr) return false
+  const url = new URL(urlStr, 'http://host')
+  return url.pathname === '/webhooks/revenuecat'
+}
+
+const parseBody = async (req: IncomingMessage) =>
+  new Promise((resolve, reject) => {
+    let body = ''
+    req.on('data', (chunk) => (body += chunk))
```

Comment:
I believe this is implicitly converting the chunk from a buffer to a string then concatenating.  I wonder if you could get some funk here depending on how the chunks break up the string bytes— would recommend collecting the buffers, and then concatenating them and turning them into a string.